protocols/messages.go

```diff
@@ -82,6 +82,7 @@ func (se *SideEffects) Merge(other SideEffects) {
 
 	se.MessagesToSend = append(se.MessagesToSend, other.MessagesToSend...)
 	se.TransactionsToSubmit = append(se.TransactionsToSubmit, other.TransactionsToSubmit...)
+	se.ProposalsToProcess = append(se.ProposalsToProcess, other.ProposalsToProcess...)
```

Comment:
😲  This is a nasty one. Can you think of any better way of protecting against this other than good unit testing? Even that seems a bit brittle, say if we added a side effect?

Reply:
That's a good question. I think we also run into a similar problem with keeping serde logic up to date. Maybe we could write a test that uses reflection to automagically to check all properties get merged correctly? Not sure how feasible that is is in golang.